### DISCUSSION TITLE
合并langchain 一直RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu! (when checking argument for argument index in method wrapper_CUDA__index_select)

### DISCUSSION THREAD

**Discussion by forgetObsession** [ymcui/Chinese-LLaMA-Alpaca#790] [Q&A]
![image](https://github.com/ymcui/Chinese-LLaMA-Alpaca/assets/25191708/af53f4dc-612a-491e-a173-2145ae0b1cfd)

**Comment by markusMM** [comment]
Hi! ^^

I can confirm this issue!

![image](https://github.com/user-attachments/assets/00402d14-9895-42c8-bab5-ad4365f5042f)

And I had the same error previously, using just  a string as a query!

Strangely enough, `chain.invoke` works perfectly!

![image](https://github.com/user-attachments/assets/8f12db3d-4815-4e6d-ad90-851f1344514d)

... even though, I should clearly either change my pre-query or prompt template or go full RAG on this (might not be the best example shooting an LLM onto)  :-P

If someone has an idea to this....
It seems to need hard coding to fully fix :-?
